virtualization-and-containers/material/exercise_vm_containers_text.md

```diff
@@ -0,0 +1,175 @@
+  - Create a new file with the name `testfile` and add it to the repository. The file must contain your GitLab username as text. Add the file to your box using the [file provisioner](https://www.vagrantup.com/docs/provisioning/file) and place it in the home directory of the `vagrant` user.
+  - We want an [additional shared folder](https://www.vagrantup.com/docs/synced-folders/basic_usage), besides `/vagrant`, in our virtual machine. The directory where the `Vagrantfile` resides, i.e. `.`, should be mounted as `/mnt/shared/` in your container.
+  - Extend the file `bootstrap.sh` for provisioning your box. It should contain commands that you would usually use on the command line like in the VirtualBox task. The script must do the follwing:
+    - Set the environment variable `ENV_TEST_VARIABLE` to have the value of your GitLab username. You can achieve this by adding `export ENV_TEST_VARIABLE=USERNAME` to the `.bashrc` file which resides in the home directory of `vagrant` user.
+    - Install `neofetch`.
+    - **Note:** In the `bootstrap.sh` you do not have to prefix commands with `sudo` since the script is executed with superuser rights when Vagrant provisions the box.
+
+    This script must be run by the [shell provisioner](https://www.vagrantup.com/docs/provisioning/shell).
+  - Rebuild your Vagrant box using the provisioning steps. Make sure that the software is installed, all files are present and the environment variable is set.
+- After checking your Vagrant box carefully, please open a merge request in the GitHub Repository [TODO: "Exercise Virtual Machines"](https://gitlab-sim.informatik.uni-stuttgart.de/sse-test-group/exercise-virtual-machines).
+  - As title choose "[`USERNAME`] Vagrant Box Provisioning"
+  - Make sure all files are up to date (`testfile`, `bootstrap.sh`, `Vagrantfile`)
+  - Add the label `Vagrant` label to the issue and assign the issue to the lecturers `jaustar` and `uekermbn`.
+  - Double-check that all files are in the repository and up to date. Check everything carefully since a CI pipeline will try to build your Vagrant recipe.
+  - If everything looks good, create the issue.
+- Check whether the CI pipeline could verify your submission. This may take some time due to the number of concurrent submission being limited. If read the error message carefully and fix your submission by creating a new commit.
+
+
+### Further Information (Vagrant)
+
+- You might want to add the option `--provision` to the `vagrant up` command if you want to rebuild a box. In case the box is not being build/rebuild, please read the [documentation about provisioning](https://www.vagrantup.com/docs/provisioning) carefully.
+- By default, Vagrant will install all files in your home directory. If Vagrant should use a different directory, you can set the environment variable `VAGRANT_HOME` to point the alternative directory. This could look like this:
```

Comment:
```suggestion
- By default, Vagrant will install all files in your current directory. If Vagrant should use a different directory, you can set the environment variable `VAGRANT_HOME` to point the alternative directory. This could look like this:
```
?

Reply:
I have to rephrase that. Yes, Vagrant creates a directory `.vagrant` in the current directory, but that is usually not too large. You will also find the directory `${HOME}/.vagrant.d` on your machine which can grow large as Vagrant stores the downloaded base images (boxes) there.

Reply:
> Review part two: Vagrant Worked for me up to a few details. Took me 40 minutes, but I had to look at your slides again (a good student would have done that already wink )

Good to hear that we are not too far off regarding time (yet). I rather feared that this part of the exercise would be too short.